hazelcast/src/main/java/com/hazelcast/client/impl/proxy/ClientMapProxy.java

```diff
@@ -2014,4 +2015,32 @@ private V computeIfPresentLocally(K key,
             }
         }
     }
+
+    @Override
+    public V computeIfAbsent(K key, Function<? super K, ? extends V> mappingFunction) {
+        checkNotNull(key, NULL_KEY_IS_NOT_ALLOWED);
+        checkNotNull(mappingFunction, NULL_FUNCTION_IS_NOT_ALLOWED);
+
+        return computeIfAbsentLocally(key, mappingFunction);
+    }
+
+    private V computeIfAbsentLocally(K key, Function<? super K, ? extends V> mappingFunction) {
```

Comment:
Should the actual logic not be done on the server side?

It seems you are first getting the value, then convert it and try to set it. 

If it is all done remotely, less remote calls are involved.

Reply:
Hi Peter,
For the client, it was decided to execute the function locally - 
https://github.com/hazelcast/hazelcast/pull/16636#discussion_r393635780